sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -68,50 +87,33 @@ def __init__(
         )
         self._eventhub_name = eventhub_client.eventhub_name
         self._event_handler = on_event
-        self._batch = kwargs.get("batch") or False
-        self._max_batch_size = kwargs.get("max_batch_size") or 300
-        self._max_wait_time = kwargs.get("max_wait_time")
-        self._partition_id: Optional[str] = kwargs.get("partition_id", None)
-        self._error_handler: Optional[Callable[[PartitionContext, Exception], None]] = kwargs.get(
-            "on_error", None
-        )
-        self._partition_initialize_handler: Optional[Callable[[PartitionContext], None]] = kwargs.get(
-            "on_partition_initialize", None
-        )
-        self._partition_close_handler: Optional[Callable[[PartitionContext, CloseReason], None]] = kwargs.get(
-            "on_partition_close", None
-        )
-        checkpoint_store: Optional[CheckpointStore] = kwargs.get(
-            "checkpoint_store"
-        )
+        self._batch = batch
+        self._max_batch_size = max_batch_size
+        self._max_wait_time = max_wait_time
+        self._partition_id = partition_id
+        self._error_handler: Optional[Callable[[PartitionContext, Exception], None]] = on_error
+        self._partition_initialize_handler: Optional[Callable[[PartitionContext], None]] = on_partition_initialize
+        self._partition_close_handler: Optional[Callable[[PartitionContext, CloseReason], None]] = on_partition_close
         self._checkpoint_store = checkpoint_store or InMemoryCheckpointStore()
-        self._initial_event_position: Union[str, int, datetime, Dict[str, Any]] = kwargs.get(
-            "initial_event_position", "@latest"
-        )
-        self._initial_event_position_inclusive: Union[bool, Dict[str, bool]] = kwargs.get(
-            "initial_event_position_inclusive", False
-        )
-
-        self._load_balancing_interval: float = kwargs.get(
-            "load_balancing_interval", 30.0
-        )
-        self._load_balancing_strategy = (
-            kwargs.get("load_balancing_strategy") or LoadBalancingStrategy.GREEDY
-        )
-        self._ownership_timeout = kwargs.get(
-            "partition_ownership_expiration_interval", self._load_balancing_interval * 6
+        self._initial_event_position: Union[int, str, datetime, Dict[str, Union[int, str, datetime]]] = initial_event_position
```

Comment:
Can you declare a type like that Dict[blah, Union[]], for some reason I thought you had to do Union[Dict[a,b], Dict[a,c]]?

Reply:
you can declare that and as long as the input matches that its okay. You are right that it should be `Mapping` as we arent modifying that value.